### ISSUE TITLE
Add sitemap url to robots.txt

### ISSUE THREAD

**Issue by ajnyga** [pkp/pkp-lib#6965]
**Describe the problem you would like to solve**
Adding the sitemap url to robots.txt would make crawling OJS/OMP/OPS sites easier.

**Describe the solution you'd like**
Specifications here: https://developers.google.com/search/docs/advanced/robots/robots_txt#google-supported-non-group-member-lines

**Who is asking for this feature?**
Google, other search engines will benefit as well.

**Comment by NateWr** [comment]
I think that hosts will often provide their own robots.txt file. How would we ensure that we are not overriding existing robots.txt settings?

I'm also concerned about this from the documentation:

> The robots.txt file must be in the top-level directory of the host

How would we know that the application is in a top-level directory? Would a robots.txt file work for an instance where some but not all journals are hosted at a TLD?

```
base_url[one] = https://site.com/one
base_url[two] = https://two.com
```

**Comment by ajnyga** [comment]
We do already have robots.txt in every application.

**Comment by ajnyga** [comment]
Probably the easiest way of handling base_url settings would be to create robots.txt dynamically like they do in WP https://developer.wordpress.org/reference/functions/do_robots/

**Comment by NateWr** [comment]
You're right, my mistake! I'm curious how WordPress does this when it's run without `mod_rewrite`. I think when we have `disable_path_info` on it might not be possible to write the `robots.txt` file dynamically. Maybe that's ok though.

Alternatively we could write the robots.txt during install, whenever a context is added or `urlPath` is edited, and at the end of each upgrade process. If we put the info between comments we could probably guard against overwriting any customization:

```
# START SITEMAPS
# These sitemaps are dynamically generated. Do not edit these lines.
sitemap: http://...
sitemap: http://...
# END SITEMAPS
```

**Comment by ajnyga** [comment]
How would the writing of the robots.txt work if you had a site with several custom journal domains?

I mean I am thinking that it should work something like this:

1. One site and one journal: site level sitemap url in robots.txt
2. One site and several journals in the same domain: site level sitemap url in robots.txt
3. One site and several journals with custom domains: just journal level sitemap url in robots.txt shown differently for each journal

**Comment by NateWr** [comment]
For cases 1 and 2 I think we can only have a static `robots.txt` file. If OJS is run with `disable_path_info` we won't be able to intercept the request. We'd also need to make sure that we weren't overriding an existing `robots.txt` file.

Cases 3 and 4 would only arise when `disable_path_info` is off, so we could dynamically generate this as you suggest. However, we don't have any way to distinguish between cases 3 and 4. This is a decision that lives outside of the software.

Maybe we just treat cases 3 and 4 like case 3?

This is the kind of feature that can not be done with our integration tests, so I think we'd want to make sure that this feature got tested with every release. I'd hate to be pointing to a bad or wrong sitemap.

**Comment by ajnyga** [comment]
hmm, we could of course choose not to automize this at all and instead allow the site manager to setup what sitemaps are presented in robots.txt. In any case I think we should first ask Google if the four(or three) cases above make sense to them.

**Comment by ajnyga** [comment]
Just a note that I do not forget about this: it would make sense to divide the sitemap based on contexts. So the main sitemap would only be a list of sub sitemaps of individual contexts.

This would also help with the size of the sitemap, see https://github.com/pkp/pkp-lib/issues/5655

**Comment by withanage** [comment]
Just a note that  the chapter   urls were added to the OMP  sitemap  during  the https://github.com/pkp/pkp-lib/issues/7003 implementation   under the PR https://github.com/pkp/omp/pull/1061

**Comment by fgnievinski** [comment]
just noting the needed robots.txt line would look like this:
```
Sitemap: https://mydomain.com/index.php/index/sitemap
````
the URL above points to the site-wide sitemap in OJS.
